### PR TITLE
add create-if-non-existent flag for document put

### PR DESCRIPTION
**Pull request by havardpe**
@vekterli please review

### CODE REVIEW

File: document/src/main/java/com/yahoo/document/DocumentPut.java [hunk 1]
```diff
@@ -67,4 +71,10 @@ public String toString() {
         return "put of document " + getId();
     }
 
+    public void setCreateIfNonExistent(boolean value) {
```
Comment (vekterli) [inline document/src/main/java/com/yahoo/document/DocumentPut.java:74]: Should we tag these as `@Beta` for now?

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/PutDocumentMessage.java [hunk 2]
```diff
@@ -129,4 +129,11 @@ public void setCondition(TestAndSetCondition condition) {
         put.setCondition(condition);
     }
 
+    public void setCreateIfNonExistent(boolean value) {
```
Comment (vekterli) [inline documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/PutDocumentMessage.java:132]: Should we tag these as `@Beta` for now?

File: documentapi/src/test/java/com/yahoo/documentapi/messagebus/protocol/test/MessagesTestBase.java [hunk 3]
```diff
@@ -94,17 +94,36 @@ private boolean fileContentIsUnchanged(String path, byte[] dataToWrite) throws I
         return Arrays.equals(existingData, dataToWrite);
     }
 
+    public interface DataTamper {
```
Comment (vekterli) [inline documentapi/src/test/java/com/yahoo/documentapi/messagebus/protocol/test/MessagesTestBase.java:97]: Maybe this should have a `@FunctionalInterface` annotation

File: documentapi/src/vespa/documentapi/messagebus/messages/putdocumentmessage.h [hunk 4]
```diff
@@ -65,7 +66,9 @@ class PutDocumentMessage : public TestAndSetMessage {
     uint64_t getSequenceId() const override;
     uint32_t getType() const override;
     string toString() const override { return "putdocumentmessage"; }
+
+    void set_create_if_non_existent(bool value) { _create_if_non_existent = value; }
+    bool get_create_if_non_existent() const { return _create_if_non_existent; }
```
Comment (vekterli) [inline documentapi/src/vespa/documentapi/messagebus/messages/putdocumentmessage.h:71]: Consider adding `noexcept` to these

**Review by vekterli — DISMISSED** [review]
:+1:

**Comment by havardpe** [comment]
@vekterli PTAL

**Review by vekterli — APPROVED** [review]

**Pull request merged**